spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -86,7 +99,7 @@ public static boolean mutableSignature(String sig) {
     private static final class ClassAnalysis {
         private static final List<String> SETTER_LIKE_PREFIXES = Arrays.asList(
                 "set", "put", "add", "insert", "delete", "remove", "erase", "clear", "push", "pop",
-                "enqueue", "dequeue", "write", "append", "replace");
+                "enqueue", "dequeue", "write", "append", "replace", "prepend", "reset", "update");
```

Comment:
I'm personally not so positive to edit these prefixes. Name methods and mutability should have no relation.
Then it is better to keep the current implementation as is, to keep backward compatibility and [bring no surprise to users](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).